src/ast/passes/printer.cpp

```diff
@@ -262,6 +262,11 @@ void Printer::visit(AssignMapStatement &assignment)
 void Printer::visit(AssignVarStatement &assignment)
 {
   std::string indent(depth_, ' ');
+
+  if (assignment.var_decl_stmt) {
+    assignment.var_decl_stmt->accept(*this);
```

Comment:
Just for the reference, this will print the variable twice - in the declaration and in the assignment:
```
# bpftrace -e 'BEGIN { let $x = 10 }' -d ast

AST after: parser
-------------------
Program
 BEGIN
  decl
   variable: $x
  =
   variable: $x
   int: 10
```
I'm personally ok with this, just highlighting it.

Perhaps we may want to add an AST printer test for that?

Reply:
> this will print the variable twice

Right, that was my intention to separate the declaration statement from the assignment.

> Perhaps we may want to add an AST printer test for that?

Good idea. Will do.

Reply:
I do think it looks nice to print out decl-assignments like this, but it's not a true representation of the AST's structure, which is what the AST printer is meant to show.

In this script we can't differentiate between the two different AST nodes by looking at the AST's debug output:
```
let $x = 1;
let $y;
$y = 1
```
```
  decl
   variable: $x :: [int64]
  =
   variable: $x :: [int64]
   int: 1 :: [int64]
  decl
   variable: $y :: [int64]
  =
   variable: $y :: [int64]
   int: 1 :: [int64]
```

For comparison, here's Clang's AST for a similar program:
```
int x = 1;
int y;
y = 1;
```
```
    |-DeclStmt <line:2:5, col:14>
    | `-VarDecl <col:5, col:13> col:9 x 'int' cinit
    |   `-IntegerLiteral <col:13> 'int' 1
    |-DeclStmt <line:3:5, col:10>
    | `-VarDecl <col:5, col:9> col:9 used y 'int'
    `-BinaryOperator <line:4:5, col:9> 'int' lvalue '='
      |-DeclRefExpr <col:5> 'int' lvalue Var 0xc824780 'y' 'int'
      `-IntegerLiteral <col:9> 'int' 1
```

Reply:
So perhaps this:
```
let $x = 1;
```
Would get changed to something like this:
```
decl
   variable: $y
   int: 1 :: [int64]
```

Reply:
```
decl
 variable: $y
 int: 1 :: [int64]
```
I think this looks reasonable - I'd vote for this one